File: ios/Classes/Platform/TextureRegistryHandler.swift
import Flutter
import Foundation

class TextureRegistryHandler: NSObject {
    private var channel: FlutterMethodChannel?
    private let registry: FlutterTextureRegistry
    private var textureManagers = [Int64: CVPixelBufferManager]()
    
    init(registry: FlutterTextureRegistry) {
        self.registry = registry
    }
    
    func startListening(messenger: FlutterBinaryMessenger) {
        channel = FlutterMethodChannel(name: "com.mangoplayer/texture", binaryMessenger: messenger)
        channel?.setMethodCallHandler(handle)
    }
    
    func stopListening() {
        channel?.setMethodCallHandler(nil)
        channel = nil
        textureManagers.values.forEach { $0.dispose() }
        textureManagers.removeAll()
    }
    
    private func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        switch call.method {
        case "registerTexture":
            let manager = CVPixelBufferManager(registry: registry)
            let textureId = registry.register(manager)
            manager.textureId = textureId
            textureManagers[textureId] = manager
            result(["success": true, "textureId": textureId])
            
        case "unregisterTexture":
            guard let args = call.arguments as? [String: Any],
                  let textureId = (args["textureId"] as? NSNumber)?.int64Value else {
                result(FlutterError(code: "invalid_arguments", message: "textureId is null", details: nil))
                return
            }
            if let manager = textureManagers.removeValue(forKey: textureId) {
                manager.dispose()
                registry.unregisterTexture(textureId)
                result(["success": true])
            } else {
                result(FlutterError(code: "invalid_arguments", message: "texture not found", details: nil))
            }
            
        default:
            result(FlutterMethodNotImplemented)
        }
    }
    
    func getTextureManager(textureId: Int64) -> CVPixelBufferManager? {
        return textureManagers[textureId]
    }
}

File: android/src/main/kotlin/com/example/mango_player/platform/TextureRegistryHandler.kt
package com.example.mango_player.platform

import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.view.TextureRegistry

class TextureRegistryHandler(
    private val textureRegistry: TextureRegistry
) : MethodChannel.MethodCallHandler {

    private var channel: MethodChannel? = null
    private val textures = mutableMapOf<Long, TextureRegistry.SurfaceTextureEntry>()

    fun startListening(messenger: BinaryMessenger) {
        channel = MethodChannel(messenger, "com.mangoplayer/texture")
        channel?.setMethodCallHandler(this)
    }

    fun stopListening() {
        channel?.setMethodCallHandler(null)
        channel = null
        textures.values.forEach { it.release() }
        textures.clear()
    }

    override fun onMethodCall(call: MethodCall, result: MethodChannel.Result) {
        when (call.method) {
            "registerTexture" -> {
                val entry = textureRegistry.createSurfaceTexture()
                textures[entry.id()] = entry
                result.success(mapOf("success" to true, "textureId" to entry.id()))
            }
            "unregisterTexture" -> {
                val textureId = call.argument<Int>("textureId")?.toLong()
                if (textureId != null) {
                    val entry = textures.remove(textureId)
                    entry?.release()
                    result.success(mapOf("success" to true))
                } else {
                    result.error("invalid_arguments", "textureId is null", null)
                }
            }
            else -> result.notImplemented()
        }
    }
    
    fun getTextureEntry(textureId: Long): TextureRegistry.SurfaceTextureEntry? {
        return textures[textureId]
    }
}
